lib/renderer/content-scripts-injector.js

```diff
@@ -11,40 +11,54 @@ const matchesPattern = function (pattern) {
 }
 
 // Run the code with chrome API integrated.
-const runContentScript = function (extensionId, url, code) {
-  const context = {}
-  require('./chrome-api').injectTo(extensionId, false, context)
-  const wrapper = `(function (chrome) {\n  ${code}\n  })`
-  const compiledWrapper = runInThisContext(wrapper, {
-    filename: url,
-    lineOffset: 1,
-    displayErrors: true
-  })
-  return compiledWrapper.call(this, context.chrome)
+const runContentScripts = function (extensionId, js) {
+  const sandbox = {}
+  require('./chrome-api').injectTo(extensionId, false, sandbox)
+  Object.assign(sandbox, global)
+  const context = vm.createContext(sandbox)
+
+  let result = null
+  for (const {url, code} of js) {
```

Comment:
I don't think we should do that, it would make the code of content scripts un-debuggable since they would all show inside one file. I'll see if I can find a way to solve the crash.